src/sig_stfl/lms/sig_stfl_lms_functions.c

```diff
@@ -264,3 +350,139 @@ int oqs_sig_stfl_lms_verify(const uint8_t *m, size_t mlen,
 	}
 }
 
+void oqs_secret_lms_key_free(OQS_SIG_STFL_SECRET_KEY *sk) {
+	if (sk == NULL) {
+		return;
+	}
+
+	//TODO: cleanup lock_key
+
+	if (sk->sig) {
+		OQS_MEM_insecure_free(sk->sig);
+		sk->sig = NULL;
+	}
+
+	if (sk->secret_key_data) {
+		oqs_lms_key_data *key_data = (oqs_lms_key_data *)sk->secret_key_data;
+		if (key_data) {
+			OQS_MEM_secure_free(key_data->sec_key, key_data->len_sec_key);
+			key_data->sec_key = NULL;
+
+			OQS_MEM_secure_free(key_data->aux_data, key_data->len_aux_data);
+		}
+
+		OQS_MEM_insecure_free(key_data->data);
+		OQS_MEM_insecure_free(key_data);
+		sk->secret_key_data = NULL;
+	}
+}
+
+/*
+ * Convert LMS secret key object to byte string
+ * Writes secret key + aux data if present
+ */
+size_t oqs_serialize_lms_key(OQS_SIG_STFL_SECRET_KEY *sk,  uint8_t **sk_key) {
+
+	oqs_lms_key_data *lms_key_data = NULL;
+	size_t key_len = 0;
+	if (sk) {
+		uint8_t *sk_key_buf = NULL;
+		lms_key_data = sk->secret_key_data;
+		if (lms_key_data->sec_key) {
+			size_t buf_size_needed = lms_key_data->len_aux_data + lms_key_data->len_sec_key;
+			key_len = buf_size_needed;
+			/* pass back serialized data */
+			if (sk_key) {
+				if (buf_size_needed) {
+					sk_key_buf = malloc(buf_size_needed * sizeof(uint8_t));
+					if (sk_key_buf) {
+
+						if (lms_key_data->len_sec_key) {
+							memcpy(sk_key_buf, lms_key_data->sec_key, lms_key_data->len_sec_key);
+						}
+
+						if (lms_key_data->len_aux_data) {
+							memcpy(sk_key_buf + lms_key_data->len_sec_key, lms_key_data->sec_key, lms_key_data->len_aux_data);
+						}
+
+						*sk_key = sk_key_buf;
+						key_len = sk->length_secret_key + lms_key_data->len_aux_data;
+					} else {
+						OQS_MEM_insecure_free(sk_key_buf);
```

Comment:
```suggestion
						key_len = 0;
```

Reply:
Thought was find out the size of the serialized data without actually doing the work.

Reply:
Should `lms_key_data->len_sec_key` be used instead of `sk->length_secret_key`, then? (Since that's the value that was passed to `memcpy`.) Or will those values always be identical?

Reply:
(I'm assuming here that `key_len` is meant to return the length of the serialized string; if it's meant to be an upper bound or some other value let me know.)

Reply:
> 

Yes, the length of the serialized string.